sw/device/tests/sim_dv/exit_test_unlocked_bootstrap.c

```diff
@@ -0,0 +1,42 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/abs_mmio.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_base.h"
+#include "sw/device/lib/dif/dif_lc_ctrl.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+#include "sw/device/lib/testing/test_framework/status.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "otp_ctrl_regs.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+bool test_main(void) {
+  dif_lc_ctrl_t lc_ctrl;
+  CHECK_DIF_OK(dif_lc_ctrl_init(
+      mmio_region_from_addr(TOP_EARLGREY_LC_CTRL_BASE_ADDR), &lc_ctrl));
+
+  dif_lc_ctrl_state_t state;
+  CHECK_DIF_OK(dif_lc_ctrl_get_state(&lc_ctrl, &state));
+
+  uint32_t rom_en = abs_mmio_read32(
+      TOP_EARLGREY_OTP_CTRL_CORE_BASE_ADDR + OTP_CTRL_SW_CFG_WINDOW_REG_OFFSET +
+      OTP_CTRL_PARAM_CREATOR_SW_CFG_ROM_EXEC_EN_OFFSET);
+
+  if (state == kDifLcCtrlStateTestUnlocked0 || (rom_en == 0)) {
+    // The test should never reach here
+    CHECK(false);
```

Comment:
that feels pretty cosmetic to me, since the failure prints out the line number already. But I can add that if you prefer.

Reply:
Sure it does print out the line number. But a more elaborate message will help dvsim's failure bucketizer sort failures more effectively, providing the triage person a quicker context. Otherwise, `CHECK(false)` failures due to different reasons will get clubbed together.

Reply:
okay i can do that.

Reply:
fixed in #15755